packages/devtools_app/lib/src/standalone_ui/vs_code/debug_sessions.dart

```diff
@@ -86,53 +73,119 @@ class _DevToolsMenu extends StatelessWidget {
-              ScreenMetaData.memory,
-              enabled: isDebug || isProfile,
-            ),
-            _devToolsButton(
-              ScreenMetaData.performance,
-            ),
-            _devToolsButton(
-              ScreenMetaData.network,
-              enabled: isDebug,
-            ),
-            _devToolsButton(
-              ScreenMetaData.logging,
-            ),
-            // TODO(dantup): Check other screens (like appSize) work embedded and
-            //  add here.
-          ],
-          child: const Text('DevTools'),
-        ),
+        if (api.capabilities.openDevToolsPage)
+          _DevToolsMenu(
+            api: api,
+            session: session,
+            isFlutter: isFlutter,
+            isDebug: isDebug,
+            isProfile: isProfile,
+          ),
       ],
     );
   }
+}
 
-  Widget _devToolsButton(
-    ScreenMetaData screen, {
-    bool enabled = true,
-  }) {
-    return IconButton(
-      onPressed: enabled
-          ? () => unawaited(api.openDevToolsPage(session.id, screen.id))
-          : null,
-      tooltip: screen.title ?? screen.id,
-      icon: Icon(screen.icon, size: actionsIconSize),
+class _DevToolsMenu extends StatelessWidget {
+  const _DevToolsMenu({
+    required this.api,
+    required this.session,
+    required this.isFlutter,
+    required this.isDebug,
+    required this.isProfile,
+  });
+
+  final VsCodeApi api;
+  final VsCodeDebugSession session;
+  final bool isFlutter;
+  final bool isDebug;
+  final bool isProfile;
+
+  @override
+  Widget build(BuildContext context) {
+    final normalDirection = Directionality.of(context);
+    final reversedDirection = normalDirection == TextDirection.ltr
+        ? TextDirection.rtl
+        : TextDirection.ltr;
+
+    Widget devToolsButton(
+      ScreenMetaData screen, {
+      bool enabled = true,
+    }) {
+      return SizedBox(
+        width: double.infinity,
+        child: TextButton.icon(
+          style: TextButton.styleFrom(
+            alignment: Alignment.centerRight,
+            shape: const ContinuousRectangleBorder(),
+          ),
+          onPressed: enabled
+              ? () => unawaited(api.openDevToolsPage(session.id, screen.id))
+              : null,
+          label: Directionality(
+            textDirection: normalDirection,
+            child: Text(screen.title ?? screen.id),
+          ),
+          icon: Icon(screen.icon, size: actionsIconSize),
+        ),
+      );
+    }
+
+    return Directionality(
+      textDirection: reversedDirection,
+      child: MenuAnchor(
+        // TODO(dantup): How to flip the menu to be anchored from the right
+        //  and expand to the left?
+        style: const MenuStyle(
+          alignment: AlignmentDirectional.bottomStart,
+        ),
+        alignmentOffset: const Offset(2, 0),
+        menuChildren: [
+          // TODO(dantup): Ensure the order matches the DevTools tab bar (if
+          //  possible, share this order).
```

Comment:
Checkout the ScreenMetaData enum in screen.dart. We can try to match the order there. If we do this, we should also add a test to `visible_screens_test.dart` that ensures that the order of DevTools screens matches the order specified in the enum.